### ISSUE TITLE
Deploy the project to AWS (phase 1)

### ISSUE THREAD

**Issue by roll** [okfn/ckanext-unhcr#254]
# Overview

An issue to track the progress of the AWS migration

# Tasks

## To deploy staging

- [x] decide on WAF usage - will be handle by UNHCR
- [x] extract items from the meeting
- [x] use `npm-check-updates` for dependency upgrades
- [x] set the retention period for log groups
- [x] add explicit no public access to storage (S3)
- [x] add explicit no public access to database (RDS)
- [x] review the S3 buckets permissions (set not public)
- [x] review the RDS setup (see - https://github.com/okfn/ckanext-unhcr/issues/245)
- [x] review how we create the `datastore_ro` user in `docker-entrypoints` 
- [x] analyze costs and reduce it!
- [x] add notifications on a failing build
- [x] move approval notifications to SNS? - just re-use the email list
- [x] review the SSH security
- [x] new resource page is broken after cloudstorage controller update / unhcr@1.0.7 rebase
- [x] downloading/previewing resources doesn't work (permission denied)
- [x] get a review from Adria (grep for `TODO` in the codebase)
- [x] get access to the staging accounts (blocked by UNHCR)
- [x] update CIDR block for VPC to `10.0.0.0/16`
- [x] deploy staging

## To deploy production

- [x] setup http -> https redirect (fix 2 load balancers situations)
- [x] decide on single/separate accounts for staging/production
- [x] finish and review the DNS configuration (we need a certificate for `ridl/ridl-uat.unhcr.org`) - waiting for UNHCR
- [x] refresh secrets
- [x] we might need to use CloudFront to deal with permissions for ckanext-cloudstorage
- [x] review usage of the local storage (rebase on the `ckanext-cloudstorage` fork to support orgs)
- [x] review and improve the documentation
- [x] deploy production

**Comment by amercader** [comment]
@roll this looks like a great plan to me!

**Comment by roll** [comment]
@chris48s 
I would move all these from for the future part to a separate issue to make the thing less massive a little bit

**Comment by chris48s** [comment]
- [ ] Continue to log downloads when we migrate to cloudstorage (see #269 )
- [ ] Copy existing database content from existing Prod/UAT sites to AWS

**Comment by chris48s** [comment]
Cheated and moved a few bits to phase 2 :see_no_evil:
